ExternalContentProviderLab/app/src/main/java/net/serkanbal/externalcontentproviderlab/CalendarRecyclerAdapter.java

```diff
@@ -0,0 +1,89 @@
+package net.serkanbal.externalcontentproviderlab;
+
+import android.content.ContentResolver;
+import android.content.ContentUris;
+import android.database.Cursor;
+import android.net.Uri;
+import android.provider.CalendarContract;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import java.text.DateFormat;
+import java.util.List;
+
+/**
+ * Created by Serkan on 28/11/16.
+ */
+
+public class CalendarRecyclerAdapter extends RecyclerView.Adapter<CalendarRecyclerAdapter.CalendarViewHolder> {
+    private List<CalendarObject> mCalendarObjectList;
+
+    public CalendarRecyclerAdapter(List<CalendarObject> calendarObjectList) {
+        mCalendarObjectList = calendarObjectList;
+    }
+
+    @Override
+    public CalendarViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+        return new CalendarViewHolder(inflater.inflate(android.R.layout.simple_list_item_2,parent,false));
+    }
+
+    @Override
+    public void onBindViewHolder(final CalendarViewHolder holder, int position) {
+        holder.mNameTextView.setText(mCalendarObjectList.get(position).getTitle());
+
+        long unixDateStamp = mCalendarObjectList.get(position).getDate();
+
+        String startTime = DateFormat.getDateInstance(DateFormat.SHORT).format(unixDateStamp);
+
+        holder.mDateTextView.setText(startTime);
+
+        holder.mNameTextView.setOnLongClickListener(new View.OnLongClickListener() {
+            @Override
+            public boolean onLongClick(View view) {
+                long id = mCalendarObjectList.get(holder.getAdapterPosition()).getId();
+                Uri uriWithId = ContentUris.withAppendedId(CalendarContract.Events.CONTENT_URI, id);
+
+                ContentResolver contentResolver = view.getContext().getContentResolver();
+                contentResolver.delete(uriWithId, null, null);
+                mCalendarObjectList.remove(holder.getAdapterPosition());
+                notifyDataSetChanged();
```

Comment:
This call to `notifyDataSetChanged()` is redundant - when the content provider is updated it will trigger the `onLoaderReset()` method you defined in your activity, which will then update the adapter